build.gradle

```diff
@@ -38,6 +38,14 @@ allprojects {
     apply plugin: 'idea'
     apply plugin: 'eclipse'
 
+    tasks.withType(JavaCompile) {
```

Comment:
It's better to set `sourceCompatibility` directly on the project, and don't set `targetCompatibility`, as it defaults to be the same as `sourceCompatibility`.

_Why?_
Because setting it on the project will result in the correct attributes being applied to outgoing configurations (used for for dependencies between projects). The attributes mark which version of Java the jar is compatible with, and this is derived from `sourceCompatibility` as set on the project.

Reply:
```
    project.pluginManager.withPlugin('java', {

        tasks.withType(JavaCompile) {
            sourceCompatibility = 1.8
        }
    })
```

@vierbergenlars what do you think of doing it like this?

Reply:
wait, that still sets the `sourceCompatibility` on the tasks. 

So it should be this:
```
    project.pluginManager.withPlugin('java', {
       sourceCompatibility = 1.8
    })
```

?

Reply:
It should be configured directly on the project, not on a `JavaCompile` task, so:

```groovy
project.pluginManager.withPlugin('java-base') {
    project.sourceCompatibility = 1.8
}
```

Reply:
I assume both `java` and `java-library` apply `java-base`?

Reply:
Yes, that is correct. And it is the `java-base` plugin that adds the `sourceCompatibility` extension to a project, so that's why you use that plugin in `withPlugin`

Reply:
Done, could you please review the changes? :)